# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1223480126 14400
#      Wed Oct 08 11:35:26 2008 -0400
# Node ID 4c85b8056d0bc57d795fa5e9f88787ca45a1db9f
# Parent  32e9e8103390532874051b0ce02a79aee279ac93
delete.m: pass array of handles to __go_delete__

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2008-10-07  John W. Eaton  <jwe@octave.org>
+
+	* miscellaneous/delete.m: Pass array of handles to __go_delete__.
+
 2008-10-06  John W. Eaton  <jwe@octave.org>
 
 	* plot/__add_datasource__.m, plot/__axes_limits__.m,
 	plot/__bar__.m, plot/__bars__.m, plot/__contour__.m,
 	plot/__go_draw_axes__.m, plot/__go_draw_figure__.m,
 	plot/__patch__.m, plot/__quiver__.m, plot/__scatter__.m,
 	plot/__stem__.m, plot/ancestor.m, plot/axis.m, plot/box.m,
 	plot/caxis.m, plot/close.m, plot/colorbar.m, plot/fill.m,
diff --git a/scripts/miscellaneous/delete.m b/scripts/miscellaneous/delete.m
--- a/scripts/miscellaneous/delete.m
+++ b/scripts/miscellaneous/delete.m
@@ -26,21 +26,19 @@
 
 ## Author: jwe
 
 function delete (arg)
 
   if (nargin == 1)
     if (ischar (arg))
       unlink (arg);
-    elseif (all (ishandle (arg)))
+    elseif (all (ishandle (arg(:))))
       ## Delete a graphics object.
-      for n = 1:numel(arg)
-         __go_delete__ (arg(n));
-      endfor
+      __go_delete__ (arg);
     else
       error ("delete: expecting argument to be a filename or graphics handle");
     endif
   else
     print_usage ();
   endif
 
 endfunction
